src/core/stokesian_dynamics/sd_interface.cpp

```diff
@@ -151,11 +151,9 @@ void set_sd_device(std::string const &dev) {
 #else
     device = INVALID;
 #endif
-  } else if (dev == "gpu") {
 #ifdef STOKESIAN_DYNAMICS_GPU
+  } else if (dev == "gpu") {
     device = GPU;
-#else
-    device = INVALID;
 #endif
   } else {
     device = INVALID;
```

Comment:
here you might run into a compiler warning in the Clang CI job with empty config file, as the conditional will lead to `device = INVALID;` in both branches

Reply:
Lol, I hoped that this wouldn't be tested on CI. Ahh this overly clever compiler!